crates/aptos-genesis/src/extra.rs

```diff
@@ -0,0 +1,212 @@
+    stake_amount: u64,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Operator {
+    operator: AccountAddress,
+    amount: u64,
+    commission: u64,
+}
+
+impl Serialize for Operator {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let vec = vec![
+            self.operator.to_string(),
+            self.amount.to_string(),
+            self.commission.to_string(),
+        ];
+        <Vec<String>>::serialize(&vec, serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for Operator {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let vec = <Vec<String>>::deserialize(deserializer)?;
+
+        if vec.len() != 3 {
+            return Err(D::Error::custom(format!(
+                "Invalid number of fields, expected {} got {}",
+                3,
+                vec.len()
+            )));
+        }
+
+        let operator = AccountAddress::from_str(vec.get(0).unwrap())
+            .map_err(|err| D::Error::custom(format!("Invalid operator field {}", err)))?;
+        let amount = u64::from_str(vec.get(1).unwrap())
+            .map_err(|err| D::Error::custom(format!("Invalid amount field {}", err)))?;
+        let commission = u64::from_str(vec.get(2).unwrap())
+            .map_err(|err| D::Error::custom(format!("Invalid commission field {}", err)))?;
+
+        Ok(Operator {
+            operator,
+            amount,
+            commission,
+        })
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct Organization {
+    admin: AccountAddress,
+    #[serde(default)]
+    operators: Vec<Operator>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct AccountGroup {
+    admin: AccountAddress,
+    operator: AccountAddress,
+    #[serde(default)]
+    accounts: Vec<AccountAddress>,
+    #[serde(default)]
+    beneficiaries: Vec<Option<AccountAddress>>,
+    stake_amount: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct Recovery {
+    #[serde(default)]
+    accounts: Vec<AccountAddress>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct AccountAmounts {
+    amounts: Vec<BTreeMap<AccountAddress, AptosCoin>>,
+}
+
+impl Serialize for AccountAmounts {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.amounts.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for AccountAmounts {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let amounts = <Vec<BTreeMap<AccountAddress, AptosCoin>>>::deserialize(deserializer)?;
+        Ok(AccountAmounts { amounts })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_initial_split() {
+        let yaml = "---
+- 0x123: 500.1234
+- 0x234: 10000.2345";
```

Comment:
why are we representing fractional / decimal here versus just the sub-unit?

Reply:
It looked like the source information I was getting has a variable amount of decimals in it.  Otherwise yes, I'd rather not use decimals